src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
+import { TodoStatus } from './constants';
+
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedStatus, setSelectedStatus] = useState<TodoStatus>(TodoStatus.ALL);
+  const [searchTerm, setSearchTerm] = useState<string>("");
+
+
+  useEffect(() => {
+  setLoading(true);
+  getTodos()
+    .then(todosFromServer => {
+      setTodos(todosFromServer);
+      setLoading(false);
+    })
+    .catch(error => {
+      console.error('Error fetching todos:', error);
+      setLoading(false);
+    });
```

Comment:
It will be better
```suggestion
    .then(setTodos)
    .catch(error => console.error('Error fetching todos:', error))
    .finally(() => setLoading(false));
```